src/test/ui/rfc-2632-const-trait-impl/issue-92111.rs

```diff
@@ -0,0 +1,29 @@
+// Regression test for #92111.
+//
+// The issue was that we normalize trait bounds before caching
+// results of selection. Checking that `impl NoDrop for S` requires
```

Comment:
```suggestion
// results of selection. Checking that `impl Tr for S` requires
```

Reply:
Nice catch! Though bors has already begun testing this PR. The fix would probably have to be separated to another PR.